src/c++/perf_analyzer/request_rate_manager.cc

```diff
@@ -109,22 +108,55 @@ RequestRateManager::ResetWorkers()
 void
 RequestRateManager::GenerateSchedule(const double request_rate)
 {
+  std::chrono::nanoseconds duration;
   std::function<std::chrono::nanoseconds(std::mt19937&)> distribution;
+
   if (request_distribution_ == Distribution::POISSON) {
     distribution = ScheduleDistribution<Distribution::POISSON>(request_rate);
+    // Poisson distribution needs to generate a full schedule so that the result
```

Comment:
Can you explain why poisson needs to generate a full schedule to make the result ore random/close to the desired rate? And what is "full schedule" in contrast to?

Reply:
I tweaked the comment some